tfexec/refresh.go

```diff
@@ -78,6 +80,38 @@ func (tf *Terraform) Refresh(ctx context.Context, opts ...RefreshCmdOption) erro
 	return tf.runTerraformCmd(ctx, cmd)
 }
 
+// RefreshJSON represents the Terraform refresh subcommand with the `-json` flag.
+// Using the `-json` flag will result in
+// [machine-readable](https://developer.hashicorp.com/terraform/internals/machine-readable-ui)
+// JSON being written to the supplied `io.Writer`.
+func (tf *Terraform) RefreshJSON(ctx context.Context, w io.Writer, opts ...RefreshCmdOption) error {
+	err := tf.compatible(ctx, tf0_15_3, nil)
+	if err != nil {
+		return fmt.Errorf("terraform apply -json was added in 0.15.3: %w", err)
+	}
+
+	tf.SetStdout(w)
+
+	cmd, err := tf.refreshJSONCmd(ctx, opts...)
+	if err != nil {
+		return err
+	}
+
+	return tf.runTerraformCmd(ctx, cmd)
+}
+
+func (tf *Terraform) refreshJSONCmd(ctx context.Context, opts ...RefreshCmdOption) (*exec.Cmd, error) {
+	cmd, err := tf.refreshCmd(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd.Args = append(cmd.Args[:3], cmd.Args[2:]...)
+	cmd.Args[2] = "-json"
```

Comment:
I'm assuming the slice contains 2 arguments for sure `[]string{"terraform", "refresh", ...}` and while it's _likely_ there will be other arguments after `refresh`, I'm not sure it's a good idea to rely on it as a change in the underlying `refreshCmd()` could cause panic here.

TL;DR Is there a strong reason for replacing the argument, instead of just appending `-json` to the end?

Reply:
Refactored to split the building of the args into flags that preceed the optional dir/plan which must come after all flags.